### PR TITLE
ocamlPackages.cooltt: init at unstable-2021-05-25

### PR DESCRIPTION
**Pull request by moni-dz**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by legendofmiracles** [comment]
Result of `nixpkgs-review pr 124581` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ocamlPackages.cooltt</li>
  </ul>
</details>

File: pkgs/development/ocaml-modules/cooltt/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  src = fetchFromGitHub {
+    owner = "RedPRL";
+    repo = "cooltt";
+    rev = "8ac06cbf7e05417d777f3ac6a471fe3576249f79";
+    sha256 = "sha256-JBLNJaRuP/gwlg8RS3cpOpzxChOVKfmFulf5HKhhHh4=";
+  };
+
+  propagatedBuildInputs = [
+    cmdliner
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/cooltt/default.nix:28]: `cmdliner` and `ppxlib` need not be propagated.

File: pkgs/development/ocaml-modules/cooltt/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    sha256 = "sha256-JBLNJaRuP/gwlg8RS3cpOpzxChOVKfmFulf5HKhhHh4=";
+  };
+
+  nativeBuildInputs = [
+    cmdliner
+    ppxlib
+  ];
+
+  propagatedBuildInputs = [
+    menhir
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/cooltt/default.nix:33]:
```suggestion
    menhirLib
```

File: pkgs/development/ocaml-modules/cooltt/default.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  src = fetchFromGitHub {
+    owner = "RedPRL";
+    repo = "cooltt";
+    rev = "8ac06cbf7e05417d777f3ac6a471fe3576249f79";
+    sha256 = "sha256-JBLNJaRuP/gwlg8RS3cpOpzxChOVKfmFulf5HKhhHh4=";
+  };
+
+  nativeBuildInputs = [
+    cmdliner
+    ppxlib
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/cooltt/default.nix:29]:
```suggestion
    menhir
    ppxlib
```

File: pkgs/development/ocaml-modules/cooltt/default.nix [hunk 4]
```diff
@@ -0,0 +1,45 @@
+{ lib
+, fetchFromGitHub
+, buildDunePackage
+, cmdliner
+, menhir
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/cooltt/default.nix:5]:
```suggestion
, menhir
, menhirLib
```

**Review by vbgl — CHANGES_REQUESTED** [review]
The `menhir` package has been refactored; thus `cooltt` now also depends on `menhirLib`.

**Review by vbgl — APPROVED** [review]

**Pull request merged**